File: CS/Demos/03Construction/Ordering.cs
namespace Demos.Construction;

class Ordering
{
  private string _b = Log("b");
  private string _a = Log("a");

  private static string _c = Log("c");
  private static string _d = Log("d");

  static Ordering()
  {
    Console.WriteLine("static constructor");
  }

  public Ordering()
  {
    Console.WriteLine("Constructor");
  }

  private static string Log(string data)
  {
    Console.WriteLine(data);
    return data;
  }

  public static void DoNothingStatic()
  {
    Console.WriteLine("DoNothingStatic");
  }

  public void DoNothingInstance()
  {
    Console.WriteLine("DoNothingInstance");
  }
}

File: VB/Demos/03Construction/Ordering.vb
Namespace Construction
  Class Ordering
    Dim _b As String = Log("b")
    Dim _a As String = Log("a")

    Shared _c As String = Log("c")
    Shared _d As String = Log("d")

    Shared Sub New()
      Console.WriteLine("Shared constructor")
    End Sub

    Public Sub New()
      Console.WriteLine("Constructor")
    End Sub

    Shared Function Log(data As String) As String
      Console.WriteLine(data)
      Return data
    End Function

    Public Shared Sub DoNothingShared()
      Console.WriteLine("DoNothingShared")
    End Sub

    Public Sub DoNothingInstance()
      Console.WriteLine("DoNothingInstance")
    End Sub
  End Class
End Namespace
